File: Lokalano-partnerstvo/Core/Entities/Vijest.cs
using System;

namespace Core.Entities
{
    public class Vijest : BaseEntity
    {
        public string Naslov { get; set; }
        public string Podnaslov { get; set; }
        public DateTime Datum { get; set; }
        public bool Fokus { get; set; }
        public string Objavio { get; set; }
        public int VijestKategorijaId { get; set; }
        public VijestKategorija VijestKategorija { get; set; }
        public string Sadrzaj { get; set; }
        public Photo Photo { get; set; }
        public int? PhotoId { get; set; }
        public string ImageUrl { get; set; }

        public void AddPhoto(string imageUrl, string fileName)
        {
            var photo = new Photo
            {
                FileName = fileName,
                PictureUrl = imageUrl
            };

            Photo = photo;
        }
        public void RemovePhoto()
        {
            if (Photo != null)
            {
                Photo = null;
            }
        }
    }
}

File: Lokalano-partnerstvo/client/src/app/shared/models/vijest.ts
export interface Vijest {
  id: number;
  naslov: string;
  podnaslov: string;
  datum: Date;
  fokus: boolean;
  objavio: string;
  vijestKategorija: string;
  vijestKategorijaId: number;
  sadrzaj: string;
  imageUrl: string;
  photo: Photo;
}

export interface Photo {
  id: number;
  imageUrl: string;
  fileName: string;
}
export interface VijestToCreate {

  naslov: string;
  podnaslov: string;
  datum: string;
  fokus: boolean;
  vijestKategorijaId: number;
  sadrzaj: string;
  imageUrl: string;
}

export class VijestFormValues implements VijestToCreate {
  naslov = '';
  podnaslov = '';
  datum = Date.now.toString();
  fokus = false;
  vijestKategorijaId: number;
  sadrzaj = '';
  imageUrl = '';

  constructor(init?: VijestFormValues) {
    Object.assign(this, init);
  }
}
